cmd/vic-machine/common/completion.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2018 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"gopkg.in/urfave/cli.v1"
+)
+
+// getNames extracts one or more names from a cli.Flag object, trimming whitespace from each.
+func getNames(flag cli.Flag) []string {
+	names := strings.Split(flag.GetName(), ",")
+
+	var trimmedNames []string
```

Comment:
Commenting on this doesn't make much sense, but I will still comment it from the future standpoint.
Since you know the lengths of trimmed names array you could preallocate it all at once, thus resulting in a less memory allocations.
```Go
trimmedNames := make([]string, len(names))
for i, name := range names {
   trimmedNames[i] = strings.Trim(name, " ")
}
```

Reply:
Looking at [this blog post](https://blog.golang.org/go-slices-usage-and-internals), it seems like I might be able to specify the desired capacity without setting a length:
```go
	trimmedNames := make([]string, 0, len(names))
	for _, name := range names {
		trimmedNames = append(trimmedNames, strings.Trim(name, " "))
	}
```

I find that more intuitive than setting a length up-front, since this way the length is always accurate (after the second iteration through the loop the length would be 2, and so on).

Would there be disadvantages to doing it this way, versus what you suggest?